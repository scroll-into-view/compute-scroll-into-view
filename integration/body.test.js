beforeAll(async () => {
  await page.goto('http://localhost:3000/integration/body')
})

describe('document.body !== document.scrollingElement edge cases', () => {
  test('should not attempt scrolling body', async () => {
    expect.assertions(3)
    const actual = await page.evaluate(() => {
      return window
        .computeScrollIntoView(document.querySelector('.target'), {
          scrollMode: 'always',
        })
        .map(window.mapActions)
    })
    expect(actual).toHaveLength(1)
    expect(actual[0]).toMatchObject({ el: 'html' })
    expect(actual).toMatchSnapshot()
  })
})
