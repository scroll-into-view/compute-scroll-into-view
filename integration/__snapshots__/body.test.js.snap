// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`document.body !== document.scrollingElement edge cases should not attempt scrolling body 1`] = `
Array [
  Object {
    "el": "html",
    "left": 0,
    "top": 750,
  },
]
`;
